output/elasticsearch/output.go

```diff
@@ -102,19 +115,22 @@ func (es *ElasticSearchOutput) Start() error {
 						Body:   bytes.NewReader(serialized),
 						OnSuccess: func(ctx context.Context, item esutil.BulkIndexerItem, res esutil.BulkIndexerResponseItem) {
 							// TODO: update metrics
+							es.healthWatcher.Success()
 							batch.Done(1)
 						},
 						OnFailure: func(ctx context.Context, item esutil.BulkIndexerItem, res esutil.BulkIndexerResponseItem, err error) {
 							// TODO: update metrics
+							es.healthWatcher.Fail(res.Error.Cause.Reason)
 							es.logger.Debugf("Failed to add items: %#v", res.Error.Cause.Reason)
 							batch.Done(1)
+
 						},
 					},
 				)
 				if err != nil {
+					es.healthWatcher.Fail(err.Error())
 					es.logger.Errorf("couldn't add to bulk index request: %v", err)
 					// This event couldn't be attempted, so mark it as finished.
-					batch.Done(1)
```

Comment:
Why remove the Done call here? IIRC it was needed for bookkeeping or else the batch won't be fully acknowledged.

Reply:
ah, probably removed that by accident...